docs/rules/no-unsafe-optional-chaining.md

```diff
@@ -0,0 +1,82 @@
+# disallow optional chaining that possibly errors (no-unsafe-optional-chaining)
+
+The optional chaining(`?.`) expression can short-circuit with `undefined`. Therefore, treating an evaluated optional chaining expression as a function, object, number, etc., can cause TypeError or unexpected result. For example:
+
+```js
+var obj = {};
```

Comment:
I think this doesn't explain the purpose of the rule, since `obj?.foo` doesn't short-circuit in this case.

`var obj = undefined;` or `var obj = null;` or `var obj;`